packages/api/src/routes/medical/network.ts

```diff
@@ -0,0 +1,44 @@
+import { Request, Response } from "express";
+import Router from "express-promise-router";
+import status from "http-status";
+import {
+  getHieDirectoryEntriesByFilter,
+  getHieDirectoryEntriesByFilterCount,
+} from "../../external/carequality/command/cq-directory/get-cq-directory-entries";
+import { requestLogger } from "../helpers/request-logger";
+import { asyncHandler } from "../util";
+import { dtoFromHieDirectoryEntry, NetworkDTO } from "./dtos/networkDTO";
+import { networkGetSchema } from "./schemas/network";
+
+import { PaginatedResponse } from "@metriport/shared";
+import { Pagination } from "../../command/pagination";
+import { paginated } from "../pagination";
+
+const router = Router();
+
+router.get(
+  "/",
+  requestLogger,
+  asyncHandler(async (req: Request, res: Response) => {
+    const params = networkGetSchema.parse(req.query);
```

Comment:
Curious why you chose to create a schema for this rather than just do whats on the get patient endpoint where we get filters using getFrom and then pass the req to paginated function which does the rest

Reply:
Good q, this is what zod is good for, and imo ((see another example where we do use zod))[https://github.com/metriport/metriport/blob/a0cd71dd51de04607d536ae27235e44919edd2af/packages/api/src/routes/medical/document.ts#L69] while it's kinda overkill here, I think it's good to have a convention that the first thing we do at the top of every api route handler is validate + parse the payloads with zod.

Reply:
I like that it helps validating query params, but I dislike that it's not clear from the code what params we're getting - one has navigate to a diff file to see a definition that's intrinsic to this endpoint, and that adds mental overhead. Maybe we can have the schema defined immediately above the endpoint?

Reply:
addressed